### PR TITLE
issue - add the user entity with the email field# 26

### PR DESCRIPTION
**Pull request by danielnottingham**
Screenshot from 2020-09-29 11-38-00
# Related Issue
- issue goes here

# Additional Info
- any additional information or context

# Checklist 
- [x] all conversation resolved?
- [x] Squashed commit with issue number
- [x] Unit Tests
- [x] Test coverage screenshot
- [x] Deploy link: https://fierce-springs-99712.herokuapp.com/

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    it { is_expected.to validate_presence_of(:email) }
+  end
+
+  describe 'field email format only' do
+    it { is_expected.to_not allow_value('/\A[^@\s]+@([^@\s]+\.)+[^@\s]+\z/').for(:email).on(:create) }
+  end
+
+  describe 'validates uniqueness of email' do
+    it { is_expected.to validate_uniqueness_of(:email) }
+  end
```
Comment (edimossilva) [inline spec/models/user_spec.rb:14]: add tests with emails typed like example@gmail.com should be valid and invalid@invalid@.com should not be valid

File: db/migrate/20200921163903_create_users.rb [hunk 2]
```diff
@@ -1,8 +1,7 @@
 class CreateUsers < ActiveRecord::Migration[6.0]
   def change
     create_table :users do |t|
```
Comment (edimossilva) [inline db/migrate/20200921163903_create_users.rb:3]: Never update a migration, always create a new one

Comment (edimossilva) [reply db/migrate/20200921163903_create_users.rb:3]: this can cause big issues

**Review by edimossilva — CHANGES_REQUESTED** [review]

File: db/migrate/20200921163903_create_users.rb (hunk 2 above)
Comment (danielnottingham) [reply db/migrate/20200921163903_create_users.rb:3]: OK, understood

File: spec/models/user_spec.rb (hunk 1 above)
Comment (danielnottingham) [reply spec/models/user_spec.rb:14]: Ok

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -12,4 +13,16 @@
   describe 'validates uniqueness of email' do
     it { is_expected.to validate_uniqueness_of(:email) }
   end
+
+  describe 'valid email typed' do
+    it 'should be valid email typed like ' do
+      email = User.new(email: 'valid@email.com')
+      expect(email).to be_valid
```
Comment (edimossilva) [inline spec/models/user_spec.rb:20]: rename `email` to user

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -12,4 +13,16 @@
...
+    end
+
+    it 'should not be valid email typed like ' do
+      email2 = User.new(email: 'invalid@invalid@.com')
+      expect(email2).to_not be_valid
```
Comment (edimossilva) [inline spec/models/user_spec.rb:25]: rename `email` to `user`
and check if user.errors contains the error that you are looking for

**Review by edimossilva — CHANGES_REQUESTED** [review]

File: spec/models/user_spec.rb (hunk 4 above)
Comment (danielnottingham) [reply spec/models/user_spec.rb]: ok

File: spec/models/user_spec.rb (hunk 3 above)
Comment (danielnottingham) [reply spec/models/user_spec.rb]: ok

**Review by edimossilva — APPROVED** [review]
Squash the commits

**Pull request merged**